eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -88,12 +88,38 @@ stages:
           ArtifactsSource: ${{ parameters.ArtifactsSource }}
           CMakeTestOptions: ${{ parameters.CMakeTestOptions }}
           CMakeSourceTestOptions: ${{ parameters.CMakeSourceTestOptions }}
+  
+  # QuickBuild runs a narrower set of fast builds. If these builds pass, 
+  # Integration can launch immediately without awaiting more build and test jobs
+  - stage: QuickBuild
```

Comment:
Is quick build the right name here? That might lead folks to adding more things here. Perhaps PrePublishBuild or something similar to be clearer of the intention.